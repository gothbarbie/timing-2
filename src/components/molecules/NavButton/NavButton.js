import React from 'react'
import styled from 'styled-components'

import Icon from 'generic/atoms/Icon/Icon'

const NavButtonStyle = styled.button`
  border: none;
  background-color: transparent;
  color: ${({ theme }) => theme.colors.primary};
  cursor: pointer;
  font-size: 3rem;
  padding: 0;
  margin: 0;
  display: flex;
  justify-content: center;
  align-items: center;

  &:hover,
  &:active {
    color: ${({ theme }) => theme.colors.gray};
  }
`

const NavButton = ({ direction = 'left', onClick, title }) => (
  <NavButtonStyle onClick={onClick} title={title}>
    {direction === 'left' && <Icon icon={['fas', 'arrow-circle-left']} />}
    {direction === 'right' && <Icon icon={['fas', 'arrow-circle-right']} />}
  </NavButtonStyle>
)

export default NavButton
