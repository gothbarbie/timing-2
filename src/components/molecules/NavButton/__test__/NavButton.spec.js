import React from 'react'
import { shallow } from 'enzyme'

import NavButton from '../NavButton'

describe('components/molecules/NavButton', () => {
  it('renders', () => {
    const props = {
      direction: 'left',
      onClick: jest.fn(),
      title: 'left',
    }
    const component = shallow(<NavButton {...props} />)

    expect(component).toMatchSnapshot()
  })
})
