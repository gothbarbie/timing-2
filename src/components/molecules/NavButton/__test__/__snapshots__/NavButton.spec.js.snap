// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/NavButton renders 1`] = `
<NavButtonStyle
  onClick={[MockFunction]}
  title="left"
>
  <Icon
    icon={
      Array [
        "fas",
        "arrow-circle-left",
      ]
    }
  />
</NavButtonStyle>
`;
